0038-http-auth-profiles/http-auth-profiles.md

```diff
@@ -0,0 +1,191 @@
+ * `aud` (audience): A claim that claim identifies the recipients that the JWT is intended for. See [Section 4.1.3 of RFC-7519](https://www.rfc-editor.org/rfc/rfc7519.html#section-4.1.3) for more details.
+ * `exp` (expiry): A claim that indicates a date/time after which the token should be considered invalid. Implementations SHOULD reject any tokens with a missing or invalid expiry claim.  See [Section 4.1.4 of RFC-7519](https://www.rfc-editor.org/rfc/rfc7519.html#section-4.1.4) for more details.
+
+##### Example Usage
+In this profile, the JWT is passed as an `Authorization` header in each HTTP request, using the [Bearer token](https://tools.ietf.org/html/rfc6750) scheme. 
+
+One example of such a Bearer token:
+
+`Authorization: Bearer eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJodHRwczovL2V4bWFwbGUuY29tIiwic3ViIjoiaHR0cHM6Ly9leGFtcGxlLmNvbS9hbGljZSIsImF1ZCI6Imh0dHBzOi8vZXhhbXBsZS5jb20vYm9iIiwiZXhwIjoxNTE2MjM5MDIyfQ.Vpa0XsMzFU95hPHhKYt8cd4HwbUz3F2Wt8tLXEkRZoU_z4xwktrfUxlrfOXZ5PxyVboDSZfNvUNhYawCx87M64JIyYDxGJbj1piu7m6F9_P3Qoi-h0bHmkk_K5x-CnFPqxBVhnWaqQFh3hzSzf0IHNJz5YMeZi4Mxgfxys-xcVKSrgel8xxMME4ns961ec49LurPpkvXIhXqsGXyBglgAe9tQBIpY-rbaq4dCLgHp6h2MFt-6h0z2HZ2B__uprRVoOOnK6s1-xQjS5wbKf7tez9u6o5ati2KjC94z8pOxG-OaCC4bW6G3aKjFwOtv4Xd-Qda1W37kda1cv4bb-qadg`. 
+
+Using the JWT specification, this token can be verified using the public-key that corresponds to the private key that signed the token. For example, the above token contains the following claims:
+
+* `iss`: https://idp.exmaple.org
+* `sub`: https://example.com/alice
+* `aud`: https://example.com/bob
+* `exp`: 1516239022
+
+The token can be verified using the following public-key, which in a production deployment can be advertised at a TLS-encrypted HTTP endpoint conforming to [RFC-7517](https://tools.ietf.org/html/rfc7517):
+ 
+ ```
+ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQC0H891JhR+Stgx81JyZeU48F4VUAS7E/OKvaVG5OjE9c+iIp2WcYFHqWjeBfrcPS1ADnTbxiKDd5D7EGWLDkBGha9H7m2hH/4PhywHomltp4Z1W7HJISzIS5JvPWFctKeKmhEekMi24yhtf44NkZg2zQijzLMQuxfaPGoW/88omtuDVaqQUmt3/Vx3v8D5ejQ2N8p7BrvpiUPQy+ZakAJf7MG0+EnaCjgnGAc9Q9wEBgMq6ifAENLne6BtQvA34jiWEIGDuD/veUwe0r0Ao/ZipZfcRJKybYNHbs5YQoxXOI2qo8qPwFrF2AJzak8+MwaiFYrDzGk8nV3e3i38RH0p test@example.com
+ ```
+
+##### Trade-off Summary
+* **Pros**
+  * Similar benefits to the `JWT_HS_256` profile, but more securely supports the issuer claim because each party signs tokens with its own private key. This allows token verifiers to know exactly who generated the token, including allowing for 3rd-party signing.
+  * Supports token expiry, which allows tokens to be generally short-lived so that peers can narrow the potential window of unauthorized usage in the event of token compromise.
+  * Allows for asymmetric key rotation without forcing a peer to change a shared-secret. 
```

Comment:
How?

Reply:
Using a combination of [RFC-7517](https://tools.ietf.org/html/rfc7517) and a pre-defined publishing location. I find Auth0's documentation and implementation to be pretty easy to understand: https://auth0.com/docs/jwks

Basically, when verifying a JWT, you can find the token-signer's public-key at some pre-defined endpoint (in Auth0's case secured by TLS). If the signing party (i.e., Auth0) wants to rotate their signing keys, they can advertise new ones.

In the case of two Connector peers using the `JWT_RS_256` profile, this is _one_ way this could be accomplished. This document doesn't go too deeply here because something like this is likely going to be different depending on whether or not two peers want to involve a 3rd party for signing purposes.